### ISSUE TITLE
respectsizehintsrule patch

### ISSUE THREAD

**Issue by apprehensions** [bakkeby/dwm-flexipatch#338]
`SIZEHINTS_RULED_PATCH` doesn't seem to work for me, i want to obey all sizehints for all windows EXCEPT st, so i don't have to use `anysize` patch for st.

**Comment by bakkeby** [comment]
I don't think that you can get around using the anysize patch for st. Any particular reason why you don't want that?

**Comment by bakkeby** [comment]
The only thing you need from the anysize patch is:

```diff
-	sizeh->height_inc = win.ch;
-	sizeh->width_inc = win.cw;
+	sizeh->height_inc = 1;
+	sizeh->width_inc = 1;
```

The rest is all about centering the content of the window depending on the remaining space, which can make it look like it is swimming inside the window when you resize it.

**Comment by apprehensions** [comment]
so is the patch gonna be implemented or not though, if not you can gracefully just close this issue.

**Comment by bakkeby** [comment]
Oh, so this issue is asking if the [respectsizehintsrule](https://github.com/bakkeby/patches/blob/master/dwm/dwm-respectsizehintsrule-6.3.diff) patch can be integrated into dwm-flexipatch? You could have made that more clear. 

If you had tried that patch out before asking then you would have found that this also does not solve your problem.

**Comment by apprehensions** [comment]
> also does not solve your problem.

i respect all sizehints except st's sizehints via rules.

**Comment by bakkeby** [comment]
In dwm you have that windows are generally in one of two states; they are either tiled, or they are floating.

A stock dwm has the following config.

```c
static const int resizehints = 1;    /* 1 means respect size hints in tiled resizals */
```

This being set to 1 means that dwm will respect a window's size hints also when tiled. Setting this value to 0 means that size hints for windows are ignored when the window is tiled. What this means is that size hints are always respected for floating windows.

When it comes to [respectsizehintsrule](https://github.com/bakkeby/patches/blob/master/dwm/dwm-respectsizehintsrule-6.3.diff) it was just an example patch showing how control that same setting on a per client basis using client rules. As in, it only affects whether size hints are respected for _tiled_ windows - size hints for floating windows are always respected.

What you are trying to do here is to explicitly disrespect size hints for a particular window when floating.
